server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java

```diff
@@ -348,13 +349,21 @@ protected double computeCost(
     }
 
     final ListenableFuture<List<Pair<Double, ServerHolder>>> resultsFuture = Futures.allAsList(futures);
-
+    final List<Pair<Double, ServerHolder>> bestServers = new ArrayList<>();
     try {
       for (Pair<Double, ServerHolder> server : resultsFuture.get()) {
         if (server.lhs < bestServer.lhs) {
           bestServer = server;
         }
       }
+
+      // Find All best servers and choose one randomly
+      for (Pair<Double, ServerHolder> server : resultsFuture.get()) {
```

Comment:
@gianm The first pass finds the best server and the 2nd pass finds all the servers which are same as the best. The only other way is to sort or have a priority list. Since the number of servers that we will be iterating over is small, 2 passes should be okay, right ?